lib/bridgetown-inline-svg/markup.rb

```diff
@@ -0,0 +1,96 @@
+# Converts the mark passed to the tag into a path & a hash of arguments.
+module BridgetownInlineSvg
+  class Markup
+    attr_reader :markup
+
+    # Separate file path from other attributes
+    PATH_SYNTAX = %r{
+      ^(?<path>[^\s"']+|"[^"]+"|'[^']+')
+      (?<params>.*)
+    }x
+
+    # Parse the first parameter in a string, giving :
+    #  [full_match, param_name, double_quoted_val, single_quoted_val, unquoted_val]
+    # The Regex works like :
+    # - first group
+    #    - match a group of characters that is alphanumeric, _ or -.
+    # - second group (non-capturing OR)
+    #    - match a double-quoted string
+    #    - match a single-quoted string
+    #    - match an unquoted string matching the set : [\w\.\-#]
+    #
+    PARAM_SYNTAX = %r{
+    ([\w-]+)\s*=\s*
+      (?:"([^"\\]*(?:\\.[^"\\]*)*)"|'([^'\\]*(?:\\.[^'\\]*)*)'|([\w.\-#]+))
+    }x
+
+    def self.parse(markup)
+      new(markup).call
+    end
+
+    def initialize(markup)
+      @markup = markup
+    end
+
+    def call
+      raise_exception! unless matched
+
+      [path, params]
+    end
+
+    private
+
+    def matched
+      @matched ||= markup.strip.match(PATH_SYNTAX)
+    end
+
+    def path
+      @path ||= matched["path"].sub(%r{^["']}, "").sub(%r{["']$}, "").strip
+    end
+
+    def params
+      @params = {}
+
+      params_parse_with_liquid_match!
+      params_parse_with_custom_match!
+      params_set_height_if_missing!
+
+      @params
+    end
```

Comment:
I wonder if we could switch this to: 

```suggestion
    def params
      @params ||= begin
        params_parse_with_liquid_match!
        params_parse_with_custom_match!
        params_set_height_if_missing!
      end
    end
```

Reply:
I added the `begin` & `end`, annoying it still needed to setup `@params` as a hash return `@params` . I'll come back to it another day :)